### ISSUE TITLE
HTTPRequest no longer returns when Accept text event-stream header set

### ISSUE THREAD

**Issue by WolfgangSenff** [godotengine/godot#26238]
**Godot version:**
Godot 3.1beta6

**OS/device including version:**
MacOS, Mojave, 10.14.3

**Issue description:**
I was working on my plugin for Firebase when I noticed, after updating to the latest beta, I no longer ever receive data when making a request via HTTPRequest with the "Accept: text/event-stream" header. This previously worked perfectly fine in Godot 3.1beta5. Also note: if you remove the header from the repro project, it also works, and this might be a work-around I can do for now.

There are quite a few changes to HTTPRequest that I saw in the commit log, but I couldn't pin down a specific one that would break this, in part because it's not actually my area of expertise...just happen to be working with it right now.

**Steps to reproduce:**
1) Find a site that supports the EventSource/Server-Sent Events protocol found here http://www.w3.org/TR/eventsource/ 
2) Make a normal scene, add an HTTPRequest node
3) Connect to the HTTPRequest node's request_completed signal
4) Make a request to your site, using the "Accept: text/event-stream" header
5) Observe that the request_completed signal is never fired

**Minimal reproduction project:**
Included is my project that connects to my Firebase realtime database. Try not to send too, too many requests, but definitely as many as you need to debug this! Note that it works fine in beta5, is only broken in beta6.

[Test3D.zip](https://github.com/godotengine/godot/files/2898040/Test3D.zip)

**Comment by WolfgangSenff** [comment]
Sorry I couldn't help more in whittling it down, @Faless; hopefully you're the right person to tag.

**Comment by WolfgangSenff** [comment]
Just tested in today's beta7, and it is still borked there as well.

**Comment by Faless** [comment]
@WolfgangSenff thank you for reporting this issue.
This is probably related to this change #26077 which superseded #25985 .
Note that the new behaviour for `HTTPClient` is correct (i.e. it now correctly keeps the connection open when it's supposed to read until EOF).
In practice, this also results in `HTTPRequests` never firing `request_complete` when using server side events as the request will not complete unless the stream is interrupted.
If you need to keep the connection open, you can use `HTTPClient` directly (and read events as they come from the server).
Instead, if you want to make periodic requests (and let each of them return) do not use server side events (i.e. remove the `Accept: text/event-stream` header in your case).
Basically `HTTPRequest` does not support server side events parsing and I don't think it's worth implementing such a special case, as it's easy to do it via script using `HTTPClient` directly.

**Comment by WolfgangSenff** [comment]
Interesting. Okay. I actually don't want to make periodic requests at all - I want SSEs to work correctly, but they weren't before this point (3.1beta5 and before), as far as I could tell, and I couldn't get HTTPClient to function correctly.

So I'll have to rewrite what I have in terms of SSEs and HTTPClient then? That's okay, just was having issues before this using HTTPClient for this particular problem. I'll go ahead and attempt it again though, see if I can make sense of it better now that I have a little more experience under my belt.

Thank you for the help, @Faless.

**Comment by Faless** [comment]
Closing since it now works as intended. See https://github.com/godotengine/godot/issues/26238#issuecomment-466819999 .
`HTTPClient` now works correctly with SSE,
`HTTPRequest` cannot work with SSE, as it only returns at the end of the stream (which is usually never in the case of SSE).
